### PR TITLE
Add `to_gpu` and `to_cpu` support for both tensor and DataFrame

### LINKED ISSUE #577: Operand's `gpu` property should record value if it's deterministic
**Is your feature request related to a problem? Please describe.**

Currently, `gpu` property will not have value unless its operand is a data source one, and `gpu` is specified. The subsequent operands' `gpu` property would be None.  It's not friendly enough for optimizer because it would not know whether optimize should be cpu or gpu specific. Thus, I suggest that  we should record true or false for `gpu` property when all inputs are on CPU or GPU.

### LINKED ISSUE #580: Add `to_gpu` and `to_cpu` support
**Is your feature request related to a problem? Please describe.**

For now, we can only specify `gpu` when creating a tensor, when we want to move a tensor to gpu if it's generated by other computation, there is no way.

**Additional context**

The `gpu` property would have value if #577 finished when all inputs are on GPU or CPU.

### PR DESCRIPTION
**Pull request by qinxuye**
## What do these changes do?

Add `to_gpu` and `to_cpu` to tensor and DataFrame.

- [x] `Tensor.to_gpu`
- [x] `Tensor.to_cpu`
- [x] `DataFrame.to_gpu`
- [x] `DataFrame.to_cpu`
- [x] `Series.to_gpu`
- [x] `Series.to_cpu`

## Related issue number

Resolves #580 .
Resolves #577 .

### CODE REVIEW

File: mars/core.py [hunk 1]
```diff
@@ -667,6 +667,16 @@ class TileableOperandMixin(object):
     def check_inputs(self, inputs):
         pass
 
+    def _check_if_gpu(self, inputs):
```
Comment (wjsi) [inline mars/core.py]: May change to ``_fill_if_gpu`` as the function changes the attributes of entities.

File: mars/dataframe/merge/concat.py [hunk 2]
```diff
@@ -14,6 +14,10 @@
 
 import pandas as pd
 import numpy as np
+try:
+    import cudf
+except ImportError:  # pragma: no cover
+    cudf = None
```
Comment (wjsi) [inline mars/dataframe/merge/concat.py]: Use ``lazy_import``, as processes forked after ``cudaInit()`` may be problematic.

Comment (qinxuye) [reply mars/dataframe/merge/concat.py]: Done.

File: mars/core.py (hunk 1 above)
Comment (qinxuye) [reply mars/core.py]: I keep the function name, and return value in it.

**Pull request merged**